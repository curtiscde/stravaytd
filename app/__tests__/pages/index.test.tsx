import React from 'react';
import { render } from '@testing-library/react';
import Home, { getStaticProps } from '../../pages/index';

const distanceData = 1;
const runsData = 2;
const movingTimeData = 3;
const elevationGainData = 4;
const year = 2023;

jest.mock('../../util/getStats', () => ({
  getStats: () => ({
    distanceData,
    runsData,
    movingTimeData,
    elevationGainData,
  }),
}));

// eslint-disable-next-line react/function-component-definition
jest.mock('../../components/YtdChart', () => () => <div>ytdchart</div>);

describe('Home', () => {
  let container: any;

  const mockData = {
    datasets: [
      {
        label: 'foo',
        borderColor: 'rgba(255, 99, 132, 1)',
        backgroundColor: 'rgba(255, 99, 132, 0.2)',
        data: [
          { x: 1646729876254, y: 1 },
          { x: 1646729876255, y: 2 },
          { x: 1646729876256, y: 3 },
        ],
      },
      {
        label: 'bar',
        borderColor: 'rgba(54, 162, 235, 1)',
        backgroundColor: 'rgba(54, 162, 235, 0.2)',
        data: [
          { x: 1646729876254, y: 2 },
          { x: 1646729876255, y: 4 },
          { x: 1646729876256, y: 6 },
        ],
      },
    ],
  };

  beforeAll(() => {
    container = render(
      <Home
        distanceData={mockData}
        movingTimeData={mockData}
        runsData={mockData}
        elevationGainData={mockData}
        year={year}
      />,
    ).container;
  });

  it('renders index page', () => {
    expect(container).toMatchSnapshot();
  });
});

describe('getStaticProps', () => {
  it('returns props', async () => {
    expect(await getStaticProps()).toEqual({
      props: {
        distanceData,
        runsData,
        movingTimeData,
        elevationGainData,
        year,
      },
    });
  });
});
