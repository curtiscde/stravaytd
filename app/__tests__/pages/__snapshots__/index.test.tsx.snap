// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders index page 1`] = `
<div>
  <header
    class="container flex items-center justify-between w-full max-w-3xl px-6 py-10 mx-auto xl:max-w-5xl"
  >
    <a
      class="block h-6 text-2xl font-semibold"
      href="/"
      rel="noopener noreferrer"
    >
      Strava Year-To-Date Stats
    </a>
    <div
      class="flex items-center"
    >
      <div
        class="hidden space-x-2 font-medium sm:block"
      >
        <a
          class="p-1"
          href="/#distance"
          rel="noopener noreferrer"
        >
          🏃‍♂️ Distance
        </a>
        <a
          class="p-1"
          href="/#runs"
          rel="noopener noreferrer"
        >
          💯 Runs
        </a>
        <a
          class="p-1"
          href="/#time"
          rel="noopener noreferrer"
        >
          ⏱ Time
        </a>
        <a
          class="p-1"
          href="/#elevation"
          rel="noopener noreferrer"
        >
          🏔 Elevation
        </a>
      </div>
    </div>
  </header>
  <div>
    <a
      id="distance"
    />
    <section
      class="dark:bg-coolGray-800 dark:text-coolGray-100"
    >
      <div
        class="container max-w-xl p-6 py-12 mx-auto space-y-24 lg:px-8 lg:max-w-7xl"
      >
        <div>
          <h2
            class="text-3xl font-bold tracking-tight text-center sm:text-5xl dark:text-coolGray-50"
          >
            🏃‍♂️ Total Distance
          </h2>
          <p
            class="max-w-3xl mx-auto mt-4 text-xl text-center dark:text-coolGray-400"
          >
            Total distance since the start of 2023
          </p>
        </div>
      </div>
    </section>
    <div>
      ytdchart
    </div>
  </div>
  <div>
    <a
      id="runs"
    />
    <section
      class="dark:bg-coolGray-800 dark:text-coolGray-100"
    >
      <div
        class="container max-w-xl p-6 py-12 mx-auto space-y-24 lg:px-8 lg:max-w-7xl"
      >
        <div>
          <h2
            class="text-3xl font-bold tracking-tight text-center sm:text-5xl dark:text-coolGray-50"
          >
            💯 Total Runs
          </h2>
          <p
            class="max-w-3xl mx-auto mt-4 text-xl text-center dark:text-coolGray-400"
          >
            Total runs since the start of 2023
          </p>
        </div>
      </div>
    </section>
    <div>
      ytdchart
    </div>
  </div>
  <div>
    <a
      id="time"
    />
    <section
      class="dark:bg-coolGray-800 dark:text-coolGray-100"
    >
      <div
        class="container max-w-xl p-6 py-12 mx-auto space-y-24 lg:px-8 lg:max-w-7xl"
      >
        <div>
          <h2
            class="text-3xl font-bold tracking-tight text-center sm:text-5xl dark:text-coolGray-50"
          >
            ⏱ Total Moving Time
          </h2>
          <p
            class="max-w-3xl mx-auto mt-4 text-xl text-center dark:text-coolGray-400"
          >
            Total moving time since the start of 2023
          </p>
        </div>
      </div>
    </section>
    <div>
      ytdchart
    </div>
  </div>
  <div>
    <a
      id="elevation"
    />
    <section
      class="dark:bg-coolGray-800 dark:text-coolGray-100"
    >
      <div
        class="container max-w-xl p-6 py-12 mx-auto space-y-24 lg:px-8 lg:max-w-7xl"
      >
        <div>
          <h2
            class="text-3xl font-bold tracking-tight text-center sm:text-5xl dark:text-coolGray-50"
          >
            🏔 Total Elevation Gain
          </h2>
          <p
            class="max-w-3xl mx-auto mt-4 text-xl text-center dark:text-coolGray-400"
          >
            Total elevation gain since the start of 2023
          </p>
        </div>
      </div>
    </section>
    <div>
      ytdchart
    </div>
  </div>
  <footer
    class="py-6 dark:bg-coolGray-800 dark:text-coolGray-50"
  >
    <div
      class="container px-6 mx-auto space-y-6 divide-y divide-coolGray-400 md:space-y-12 divide-opacity-50"
    >
      <div
        class="grid justify-center pt-6 lg:justify-between"
      >
        <div
          class="flex flex-col self-center text-sm text-center md:block lg:col-start-1 md:space-x-6"
        >
          <a
            href="/#distance"
            rel="noopener noreferrer"
          >
            <span>
              🏃‍♂️ Distance
            </span>
          </a>
          <a
            href="/#runs"
            rel="noopener noreferrer"
          >
            <span>
              💯 Runs
            </span>
          </a>
          <a
            href="/#time"
            rel="noopener noreferrer"
          >
            <span>
              ⏱ Time
            </span>
          </a>
          <a
            href="/#elevation"
            rel="noopener noreferrer"
          >
            <span>
              🏔 Elevation
            </span>
          </a>
        </div>
        <div
          class="flex justify-center pt-4 space-x-4 lg:pt-0 lg:col-end-13"
        >
          <a
            class="flex items-center justify-center w-40 h-10 rounded-full dark:text-coolGray-900"
            href="https://www.strava.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Powered by Strava"
              src="/images/api_logo_pwrdBy_strava_horiz_light.png"
            />
          </a>
          <a
            class="flex items-center justify-center w-10 h-10 rounded-full dark:bg-teal-400 dark:text-coolGray-900"
            href="https://www.curtiscode.dev"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.51211712,15 L8.17190229,15 C8.05949197,14.0523506 8,13.0444554 8,12 C8,10.9555446 8.05949197,9.94764942 8.17190229,9 L3.51211712,9 C3.18046266,9.93833678 3,10.9480937 3,12 C3,13.0519063 3.18046266,14.0616632 3.51211712,15 L3.51211712,15 Z M3.93551965,16 C5.12590433,18.3953444 7.35207678,20.1851177 10.0280093,20.783292 C9.24889451,19.7227751 8.65216136,18.0371362 8.31375067,16 L3.93551965,16 L3.93551965,16 Z M20.4878829,15 C20.8195373,14.0616632 21,13.0519063 21,12 C21,10.9480937 20.8195373,9.93833678 20.4878829,9 L15.8280977,9 C15.940508,9.94764942 16,10.9555446 16,12 C16,13.0444554 15.940508,14.0523506 15.8280977,15 L20.4878829,15 L20.4878829,15 Z M20.0644804,16 L15.6862493,16 C15.3478386,18.0371362 14.7511055,19.7227751 13.9719907,20.783292 C16.6479232,20.1851177 18.8740957,18.3953444 20.0644804,16 L20.0644804,16 Z M9.18440269,15 L14.8155973,15 C14.9340177,14.0623882 15,13.0528256 15,12 C15,10.9471744 14.9340177,9.93761183 14.8155973,9 L9.18440269,9 C9.06598229,9.93761183 9,10.9471744 9,12 C9,13.0528256 9.06598229,14.0623882 9.18440269,15 L9.18440269,15 Z M9.3349823,16 C9.85717082,18.9678295 10.9180729,21 12,21 C13.0819271,21 14.1428292,18.9678295 14.6650177,16 L9.3349823,16 L9.3349823,16 Z M3.93551965,8 L8.31375067,8 C8.65216136,5.96286383 9.24889451,4.27722486 10.0280093,3.21670804 C7.35207678,3.81488234 5.12590433,5.60465556 3.93551965,8 L3.93551965,8 Z M20.0644804,8 C18.8740957,5.60465556 16.6479232,3.81488234 13.9719907,3.21670804 C14.7511055,4.27722486 15.3478386,5.96286383 15.6862493,8 L20.0644804,8 L20.0644804,8 Z M9.3349823,8 L14.6650177,8 C14.1428292,5.03217048 13.0819271,3 12,3 C10.9180729,3 9.85717082,5.03217048 9.3349823,8 L9.3349823,8 Z M12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 Z"
              />
            </svg>
          </a>
          <a
            class="flex items-center justify-center w-10 h-10 rounded-full dark:bg-teal-400 dark:text-coolGray-900"
            href="https://github.com/curtiscde"
            rel="noopener noreferrer"
            target="_blank"
            title="GitHub"
          >
            <svg
              class="w-5 h-5"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.9,2.1c-4.6,0.5-8.3,4.2-8.8,8.7c-0.5,4.7,2.2,8.9,6.3,10.5C8.7,21.4,9,21.2,9,20.8v-1.6c0,0-0.4,0.1-0.9,0.1 c-1.4,0-2-1.2-2.1-1.9c-0.1-0.4-0.3-0.7-0.6-1C5.1,16.3,5,16.3,5,16.2C5,16,5.3,16,5.4,16c0.6,0,1.1,0.7,1.3,1c0.5,0.8,1.1,1,1.4,1 c0.4,0,0.7-0.1,0.9-0.2c0.1-0.7,0.4-1.4,1-1.8c-2.3-0.5-4-1.8-4-4c0-1.1,0.5-2.2,1.2-3C7.1,8.8,7,8.3,7,7.6C7,7.2,7,6.6,7.3,6 c0,0,1.4,0,2.8,1.3C10.6,7.1,11.3,7,12,7s1.4,0.1,2,0.3C15.3,6,16.8,6,16.8,6C17,6.6,17,7.2,17,7.6c0,0.8-0.1,1.2-0.2,1.4 c0.7,0.8,1.2,1.8,1.2,3c0,2.2-1.7,3.5-4,4c0.6,0.5,1,1.4,1,2.3v2.6c0,0.3,0.3,0.6,0.7,0.5c3.7-1.5,6.3-5.1,6.3-9.3 C22,6.1,16.9,1.4,10.9,2.1z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
